@here/harp-mapview/lib/MapView.ts

```diff
@@ -2574,18 +2571,63 @@ export class MapView extends THREE.EventDispatcher {
         this.m_screenCollisions.update(width, height);
 
         this.m_pixelToWorld = undefined;
+        this.m_fog.update(this, this.m_viewRanges.maximum);
 
+        this.computeTargetAndDistance();
+        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(this, this.m_targetDistance);
+    }
+
+    private computeTargetAndDistance() {
         const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
```

Comment:
This method is HIGHLY in-practical and over complicated for getting something like camera pitch, we may consider creating or using other method here (maybe something like extractCameraTilt()?).

Reply:
Yes, this is just the first step. Eventually a lot of methods in MapViewUtils will be removed and either moved to MapView as private or completely removed.
My ultimate goal is to have one `MapViewState` object that is private in `MapView` and that wuill contain all necessary information (e.g. tilt, heading, target, targetDistance, ...). With this aproach we could also heavily reduce the public interface of `MapView` (e.g. nobody need `targetDistance` outside of `MapView`)

Reply:
Ok, understood... although keep in mind that MapViewState may be still inaccessible in few components may be unlinked from MapView - dependency removed:
- ClipPlanesEvaluator(s),
- ZoomLevelEvaluator(s),
- new one - MapViewAtmosphere.

Reply:
The MapViewState object will be a private member of MapView, but it can still be passed to the components that need it.
So instead of having the components pull what they need from MapView (hence the MapView API explodes) MapView would rather push the necessary properties to the components (like you did already for the ClipPlaneEvaluator).